cws-service/src/main/java/jpl/cws/controller/RestService.java

```diff
@@ -299,7 +301,14 @@ public ModelAndView getInitiatorsHtmlTable() {
 	public @ResponseBody String deployModelerFile(
 			@RequestParam("filename") String filename,
 			@RequestParam("xmlData") String xmlData) {
-		
+
+		// Don't allow filename to contain any special characters
+		Pattern p = Pattern.compile("^[\\w-]*$");
```

Comment:
This regex only allows alpha-numeric, underscores (_), and hyphens (-).  So a filename with an extension will be rejected (e.g. foo.bar).  Is filename here assumed to not have an extension?  In general a dot should be allowed, but not a double dot preceded and/or followed by slash.

Reply:
This filename doesn't include the extension.  That is appended on later.